src/db/db.py

```diff
@@ -0,0 +1,109 @@
+from datetime import datetime
+from models import Address, Block, UTXO
+from mongoengine import Document
+from rpc import get_block, get_block_count
+import json
+
+db_host = 'mongodb://mongo:27017/blockchain'
+db_name = 'blockchain'
+
+def drop_db():
+    Block.drop_collection()
+    Address.drop_collection()
+    UTXO.drop_collection()
+    
+def _set_insert_or_update_time(db_object, ins=True, upd=True):
+    time = datetime.now().timestamp()
+    
+    if ins:
+        db_object.insert_time = time
+    if upd:
+        db_object.update_time = time
+    
+def get_highest_block_number_in_db():
+    if Block.objects().count() == 0:
+        return -1
+    return Block.objects().order_by('-height').limit(1)[0].height
+
+def _create_utxo(tx_key, addr, amount):
+    json_utxo = json.dumps({'id':tx_key, 'address':addr, 'value':amount})
+    return UTXO.from_json(json_utxo)
+    
+def _parse_inputs(txid, vin, utxo, new_utxo):
+    for inp in vin:
+        if 'coinbase' in inp:
+            continue
+        in_txid = inp.txid
+        in_vout = inp.vout
+        tx_key = in_txid + str(in_vout)
+        try: 
+            db_utxo = utxo.get(id=tx_key)
+            if db_utxo.id in new_utxo:
+                del new_utxo[db_utxo.id]
+            db_utxo.delete()
+            print('Deleted utxo: ' + db_utxo.to_json())
+        except:
+            exception_string = "Invalid input. txid: " + str(txid) + ", vin_txid: " + str(
+                in_txid) + ", vout: " + str(in_vout)
+            raise BaseException(exception_string)
```

Comment:
This exception needs to be handled

Reply:
Fixed by workaround which resets DB on every daemon restart.